low-level/actuators/impl/centos_7/systemd_service.py

```diff
@@ -80,44 +82,59 @@ def perform_request(self, jsonMsg):
             self._proxy = self._bus.get_object("org.freedesktop.systemd1", str(systemd_unit))
 
             # The returned result of the desired action
-            result = None
-
-            if self._service_request == "restart":
-                self._manager.RestartUnit(self._service_name, 'replace')
-
-                # Ensure we get an "active" status and not "activating"
-                max_wait = 0
-                while self._get_activestate() != "active":
-                    self._log_debug("status is still activating, pausing")
-                    time.sleep(1)
-                    max_wait += 1
-                    if max_wait > 20:
-                        self._log_debug("maximum wait for service restart reached")
-                        break
-
-            elif self._service_request == "start":
-                self._manager.StartUnit(self._service_name, 'replace')
+            result = {}
+            is_err_response = False
+            if self._service_request in ['restart', 'start']:
+                # Before restart/start the service, check service state. If it is not active
+                # or activating then only process restart/start request.
+                service_state  = Service('dbus').get_state(self._service_name)

```

Comment:
Shouldn't we create one object of Service() class & reuse that further. currently new obj is created for every operation

Reply:
Done , Created self._service object at time of init.